homeassistant/components/local_calendar/manifest.json

```diff
@@ -3,6 +3,7 @@
   "name": "Local Calendar",
   "codeowners": ["@allenporter"],
   "config_flow": true,
+  "dependencies": ["diagnostics"],
```

Comment:
this shouldn't be necessary.

Reply:
Without it the tests fail with the frozen router problem, so I thought maybe I needed it. I've read this is when rest and websocket are both used in tests. I can't remember how I worked around this last time... Can you remind me what I'm doing wrong?

Reply:
Can't you setup diagnostics in a fixture at the top of the test_diagnostics file ?

```python
@pytest.fixture(autouse=True)
async def setup_diag(hass):
    assert await async_setup_component(hass, "diagnostics", {})
```

Reply:
ok, great. To answer your question: It's not that I couldn't, it just wasn't obvious this is the best practice. Thanks for clarifying!